meta-error-handling/static_assert.md

```diff
@@ -0,0 +1,113 @@
+# Module name: `static_assert`
+
+## Overview
+
+`static_assert` is a compile-time evaluated function that asserts the 
+truth of a supplied predicate, issuing an optional user-supplied error 
+message if the predicate is `false`.
+
+<table>
+  <thead>
+    <th>Level</th>
+    <th>Objectives</th>
+  </thead>
+  <tr>
+    <td>Foundational</td>
+    <td>Calling `static_assert` with a constant expression</td>
+  </tr>
+  <tr>
+    <td>Main</td>
+    <td>Using `static_assert` to detect contract violations and improve error messages</td>
+  </tr>
+  <tr>
+    <td>Advanced</td>
+    <td></td>
+  </tr>
+</table>
+
+## Motivation
+
+`static_assert` allows the developer to enforce that conditions which can 
+be checked during compilation will force build errors when violated. 
+Additionally, they are the best mechanism by which a developer can pass 
+useful information to other developers regarding what violation occured or 
+what must be done, instead.
+
+## Topic introduction
+
+## Foundational: Calling `static_assert` with a constant expression
+
+### Background/Required Knowledge
+
+A student:
+
+* Should be able to explain the difference between code evaluated at compile-time and run-time
+* Should be able to cite some examples of compile-time known information, such as `sizeof(T)`
+
+### Student outcomes
+
+A student should be able to:
+
+1. Assert the expected size of a structure using `static_assert`
+
+### Caveats
+
+### Points to cover
+
+* In addition to what is wrong, a good error message will inform the user of how to correct it
+
+## Main: Contracts and `static_assert`
+
+### Background/required knowledge
+
+* All of the above.
+* General understanding of compile-time requirements
+
+### Student outcomes
+
+A student should be able to:
+
+1. Test pre-conditions of a meta-function
+2. Test meta-functions for known values
```

Comment:
"Test" on its own is simultaneously really specific and vague. "Test" works here because of context (and [Grice's Maxims][1]), but I'm concerned it doesn't really put a bigger picture into context (see the QLD and ACT cognitive verbs below, for their definition of "test" across curricula). Perhaps I'm thinking to granularly or too standardesey.

I'd like to see SG20 adopt a set of standardised key words. Till such a time that I write the proposal and SG20 discusses it, I've been using cognitive verbs from Australian states' curricula as a prop.

* [New South Wales][2]
* [Queensland glossary][3] and [Queensland classification][4]
* [Australian Capital Territory][5] (see Appendix D)

[1]: https://youtu.be/IJEaMtNN_dM
[2]: https://educationstandards.nsw.edu.au/wps/portal/nesa/11-12/hsc/hsc-student-guide/glossary-keywords
[3]: https://www.qcaa.qld.edu.au/downloads/senior-qce/common/snr_glossary_cognitive_verbs.pdf
[4]: https://www.qcaa.qld.edu.au/downloads/senior-qce/common/snr_categories_cognitive_verbs.pdf
[5]: http://www.bsss.act.edu.au/__data/assets/word_doc/0018/314370/AC_Mathematical_Applications_14-20_-updated.docx

Reply:
updated language to mention 'utilizing static_assert'. I like it a bit better (Also loved the video) but can you check to see if it captures your intent?

Reply:
Sorry, I can't see this change. Can you please link to the line here?

Reply:
I tagged you where the updated language it. If you're coming to this from the "conversation" tab, it will show you the outdated language and marks it as "outdated" above